storybook/cypress/e2e/modal.cy.ts

```diff
@@ -29,6 +28,16 @@ describe("Modal", () => {
     cy.wait(200).matchImageSnapshot();
   });
 
+  it.only("should have the propper role set by modal component", () => {
+    cy.get("dso-modal")
+      .invoke("attr", "component-role", "alert")
+      .shadow()
+      .find(".dso-modal")
+      .should("have.attr", "role", "alert")
+      .and("have.attr", "aria-modal", "true")
+      .and("have.attr", "aria-labelledby");
+  });
+
```

Comment:
Ik zou er dit van maken:

```suggestion
  it("should have the propper role set by modal component", () => {
    cy.get("dso-modal.hydrated")
      .invoke("prop", "componentRole", "alert")
      .get("@dsoModal")
      .find(".dso-modal")
      .should("have.attr", "role", "alert");
  });

```

Reply:
fixed

Reply:
> Waarom de hydrated class gebruiken en niet de shadowDom?

Stencil zet de class `hydrated` wanneer het process van `[hydration](https://en.wikipedia.org/wiki/Hydration_(web_development))` klaar is. De class `hydrated` is de default class die stencil hier voor gebruikt Zie: https://stenciljs.com/docs/config#hydratedflag.

Voor het welslagen van veel, zo niet alle, Cypress e2e tests, willen we zeker weten dat eventuele scripting gerelateerde initialisatie van het eb-component onder test is afgerond en dus query-en we altijd met de class `.hydrated` in de component selector.

De shadowDom gebruiken we ook door de alias `@dsoModal` te gebruiken. In de beforeEach van een describe doen we vaak:

```
      .shadow()
      .as("dsoModal");
```